packages/react-components/src/dot/src/Dot.tsx

```diff
@@ -65,6 +65,30 @@ export function InnerDot(props: InnerDotProps) {
     );
 }
 
+function useColorStyle(color: string) {
+    return useMemo(() => {
+        let dotCSSDeclaration = {};
+
+        if (color) {
+            if (color.startsWith("rgb") || color.startsWith("hsl") || color.startsWith("#")) {
+                dotCSSDeclaration = {
+                    "--o-ui-dot-color": color
+                };
+            } else if (color.startsWith("--")) {
+                dotCSSDeclaration = {
+                    "--o-ui-dot-color": `var(${color})`
+                };
+            } else {
+                dotCSSDeclaration = {
+                    "--o-ui-dot-color": `var(--o-ui-global-${color})`
+                };
+            }
+        }
+
+        return dotCSSDeclaration;
+    }, [color]);
+}
```

Comment:
I would personally rename this to `useColor` and return the actual color value instead of a style object. The consumer can take care of building the style object if this is the intended use.

This way, it is more generic and less specific to a given context. If needed, this code could then be used elsewhere.

Reply:
If need it twice this would be moved in an utility component.